### PR TITLE
Sitecore 9.3 support

### PR DESCRIPTION
**Pull request by hopkirk**
## Summary

Upgrade connector to support Sitecore 9.3.

# Reworks the Following

- Upgrade Visual Studio project to Sitecore 9.3 packages.
- Upgrade Visual Studio project to .Net 4.8
- Upgrade HTMLAgilityPack version to 1.4.9.5
- Rework method for accessing SiteInfo configuration data.

### CODE REVIEW

File: src/Cognifide.Sitecore.Crownpeak/App_Config/Include/Unicorn/Unicorn.AutoPublish.config [hunk 1]
```diff
@@ -7,8 +7,8 @@
 
 	http://github.com/kamsar/Unicorn
 -->
-<configuration xmlns:patch="http://www.sitecore.net/xmlconfig/">
-	<sitecore>
+<configuration xmlns:patch="http://www.sitecore.net/xmlconfig/" xmlns:role="http://www.sitecore.net/xmlconfig/role/">
```
Comment (hopkirk) [inline src/Cognifide.Sitecore.Crownpeak/App_Config/Include/Unicorn/Unicorn.AutoPublish.config:10]: Unicorn upgrade adjusting XML namespaces.

File: src/Cognifide.Sitecore.Crownpeak/App_Config/Include/Unicorn/Unicorn.Configs.Default.example [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 	See Unicorn.config for commentary on how configurations operate, or https://github.com/kamsar/Unicorn/blob/master/README.md
 -->
-<configuration xmlns:patch="http://www.sitecore.net/xmlconfig/">
+<configuration xmlns:patch="http://www.sitecore.net/xmlconfig/" xmlns:role="http://www.sitecore.net/xmlconfig/role/">
```
Comment (hopkirk) [inline src/Cognifide.Sitecore.Crownpeak/App_Config/Include/Unicorn/Unicorn.Configs.Default.example:11]: Unicorn upgrade adjusting XML namespaces.

File: src/Cognifide.Sitecore.Crownpeak/App_Config/Include/Unicorn/Unicorn.DataProvider.config [hunk 3]
```diff
@@ -9,8 +9,8 @@
 
 	http://github.com/kamsar/Unicorn
 -->
-<configuration xmlns:patch="http://www.sitecore.net/xmlconfig/">
-	<sitecore>
+<configuration xmlns:patch="http://www.sitecore.net/xmlconfig/" xmlns:role="http://www.sitecore.net/xmlconfig/role/">
```
Comment (hopkirk) [inline src/Cognifide.Sitecore.Crownpeak/App_Config/Include/Unicorn/Unicorn.DataProvider.config:12]: Unicorn upgrade adjusting XML namespaces.

File: src/Cognifide.Sitecore.Crownpeak/App_Config/Include/Unicorn/Unicorn.Dilithium.config.example [hunk 4]
```diff
@@ -25,8 +25,8 @@
 
 	http://github.com/kamsar/Unicorn
 -->
-<configuration xmlns:patch="http://www.sitecore.net/xmlconfig/">
-	<sitecore>
+<configuration xmlns:patch="http://www.sitecore.net/xmlconfig/" xmlns:role="http://www.sitecore.net/xmlconfig/role/">
```
Comment (hopkirk) [inline src/Cognifide.Sitecore.Crownpeak/App_Config/Include/Unicorn/Unicorn.Dilithium.config.example:28]: Unicorn upgrade adjusting XML namespaces.

File: src/Cognifide.Sitecore.Crownpeak/App_Config/Include/Unicorn/Unicorn.PowerShell.config [hunk 5]
```diff
@@ -22,8 +22,8 @@
 	
 	http://github.com/kamsar/Unicorn
 -->
-<configuration xmlns:patch="http://www.sitecore.net/xmlconfig/">
-	<sitecore>
+<configuration xmlns:patch="http://www.sitecore.net/xmlconfig/" xmlns:role="http://www.sitecore.net/xmlconfig/role/">
```
Comment (hopkirk) [inline src/Cognifide.Sitecore.Crownpeak/App_Config/Include/Unicorn/Unicorn.PowerShell.config:25]: Unicorn upgrade adjusting XML namespaces.

File: src/Cognifide.Sitecore.Crownpeak/App_Config/Include/Unicorn/Unicorn.UI.IdentityServer.config.disabled [hunk 6]
```diff
@@ -0,0 +1,26 @@
+<!--
```
Comment (hopkirk) [inline src/Cognifide.Sitecore.Crownpeak/App_Config/Include/Unicorn/Unicorn.UI.IdentityServer.config.disabled:1]: Sample file included in Unicorn upgrade.

File: src/Cognifide.Sitecore.Crownpeak/App_Start/RazorGeneratorMvcStart.cs [hunk 7]
```diff
@@ -0,0 +1,21 @@
+using System.Web;
```
Comment (hopkirk) [inline src/Cognifide.Sitecore.Crownpeak/App_Start/RazorGeneratorMvcStart.cs]: Added automatically in upgrade to Sitecore 9.3 packages.

File: src/Cognifide.Sitecore.Crownpeak/Logic/SiteHelper.cs [hunk 8]
```diff
@@ -13,8 +14,8 @@ public static class SiteHelper
         public static IEnumerable<SiteInfo> Sites
         {
             get
-            {
-                return _sites ?? (_sites = global::Sitecore.Configuration.Settings.Sites.Where(s => s.EnablePreview).ToList());
+            {                
+                return _sites ?? (_sites = Factory.GetSiteInfoList().Where(s => s.EnablePreview).ToList());
```
Comment (hopkirk) [inline src/Cognifide.Sitecore.Crownpeak/Logic/SiteHelper.cs:18]: New method of accessing SiteInfo configuration data.

File: src/Cognifide.Sitecore.Crownpeak/Web.config [hunk 9]
```diff
@@ -4,12 +4,32 @@
...
+
+    The following attributes can be set on the <httpRuntime> tag.
+      <system.Web>
+        <httpRuntime targetFramework="4.8" />
+      </system.Web>
+  -->
   <system.web>
-    <compilation debug="true" targetFramework="4.5.2" />
-    <httpRuntime targetFramework="4.5.2" />
+    <compilation debug="true" targetFramework="4.8" />
```
Comment (hopkirk) [inline src/Cognifide.Sitecore.Crownpeak/Web.config:16]: Solution has been upgraded to Sitecore 9.3 / .Net 4.8.

**Review by blakejolley — APPROVED** [review]
Approving on the basis that someone knows what we are doing with regards to versions and branches

**Review by jboast — APPROVED** [review]

**Comment by girichards** [comment]
closing as this PR now forms the new `release-9.3` branch

**Pull request closed without merging**